### ISSUE TITLE
Clarification on valid path elements

### ISSUE THREAD

**Issue by srebhan** [openconfig/reference#209]
With [Telegraf](https://www.influxdata.com/time-series-platform/telegraf/)  we came across a device that uses spaces in path elements (see the [corresponding issue](https://github.com/influxdata/telegraf/issues/15405)), now the question arises if this (` show version`) is a valid path element.

When reading the [GNMI specification](https://github.com/openconfig/reference/blob/master/ietf/drafts/draft-openconfig-rtgwg-gnmi-spec-01.txt) there is no clear statement on how a path element is defined. In section 3.1 it states

>  gNMI is often used to carry payloads that contain data instances of  YANG schemas [...]

and this somehow suggests that path elements should correspond to YANG model identifiers. Those identifiers are defined in [RFC 6020 section 6.2](https://tools.ietf.org/html/rfc6020#section-6.2) as

>    Each identifier starts with an uppercase or lowercase ASCII
   letter or an underscore character, followed by zero or more ASCII
   letters, digits, underscore characters, hyphens, and dots.

i.e. there is no space allowed.

So the question is, MUST the path elements follow the YANG model identifiers specification or not? If not, what are allowed characters for path elements?

**Comment by hellt** [comment]
Yes, using cli origin with gNMI for running show commands is an uncharted territory. It is not defined in the spec (but may work with more than a couple of vendors already).

You can have a look at https://github.com/openconfig/gnmic where we made sure we are not limited by the underlying gnxi library when doing path parsing. In other words, your request would work.

**Comment by srebhan** [comment]
Thanks for the clarification everyone! Much appreciated! Just to be sure I got you right, you are saying that `cli` is not specified as a valid origin?

@hellt we are currently using the `gnxi` library and if possible we would like to stick with it... I will contact them and point them to this issue though. ;-)
